gitdiff/patch_header.go

```diff
@@ -369,3 +373,61 @@ func parseHeaderMail(mailLine string, r io.Reader) (*PatchHeader, error) {
 
 	return h, nil
 }
+
+func cleanupSubject(s string) string {
+	r := []rune(s)
```

Comment:
You should be able to index into the string bytes directly (as if it was a `[]byte`) since the loop is only concerned with plain ASCII characters. This will be consistent with how the other parsing in the library works.